src/lsp/lp_lsp.ml

```diff
@@ -166,13 +166,41 @@ let get_textPosition params =
   let line, character = int_field "line" pos, int_field "character" pos in
   line, character
 
+let closest_before (line, pos) (objs: ('a * Pos.popt) list) =
```

Comment:
Thanks to add a comment before explaining what this function is intended to do.

Reply:
done